swarm/src/connection/handler_wrapper.rs

```diff
@@ -440,3 +440,84 @@ pub enum Event<TOutboundOpenInfo, TCustom> {
     /// Other event.
     Custom(TCustom),
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::handler::PendingConnectionHandler;
+    use quickcheck::*;
+    use std::num::NonZeroU8;
+    use std::sync::Arc;
+
+    struct DummySubstream(Arc<()>);
```

Comment:
```suggestion
    struct PendingSubstream(Arc<()>);
```

Reply:
Plus, I'd prefer such kind of infrastructure to be at the bottom of the module. More generally, items in a module / file should be sorted by priority / likelihood of wanting to see them as you open the file and read it top to bottom.

Having imports at the top somewhat contradicts this rule but I think that is too controversial to adopt :D

Reply:
:+1: 62f5e4c9